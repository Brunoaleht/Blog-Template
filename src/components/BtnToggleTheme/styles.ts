import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
  `}
`;
export const Label = styled.label`
  ${() => css`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    overflow: hidden;
    font-size: 0;
    color: rgba(0, 0, 0, 0);
  `}
`;
export const BtnInput = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;
    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
    }
    &:focus + ${Slider} {
      border: solid 1px ${theme.colors.secondary};
    }
    &:checked + ${Slider}::before {
      transform: translateX(1.95rem);
      background: ${theme.colors.primary};
    }
  `}
`;
export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    cursor: pointer;
    background: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;
    border: solid 1px ${theme.colors.darkerGray};
    border-radius: 2rem;
    &::before {
      content: '';
      position: absolute;
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.1rem;
      background: ${theme.colors.textDark};
      border-radius: 50%;
      transition: all 300ms ease-in-out;
    }
  `}
`;
