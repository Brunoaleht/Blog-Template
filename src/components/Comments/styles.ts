import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
