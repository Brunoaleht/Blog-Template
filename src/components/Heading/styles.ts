import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
  `,
};

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

export const Title = styled.h1<Pick<HeadingProps, 'colorDark' | 'size' | 'uppercase'>>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primary : theme.colors.white};
    line-height: 1.2;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
