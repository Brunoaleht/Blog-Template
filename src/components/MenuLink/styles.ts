import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.white};
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 300ms ease-in-out;
    &:hover {
      border-right: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;
