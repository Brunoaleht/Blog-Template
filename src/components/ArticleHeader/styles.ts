import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';
export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.large};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.fonts.sizes.medium};
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
