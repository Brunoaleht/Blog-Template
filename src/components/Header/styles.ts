import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.textDark};
    font-size: ${theme.fonts.sizes.small};
    margin: 0 auto;
    ${HeadingStyle} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }
    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeadingStyle} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
    p {
      margin-top: 0;
    }
    @media ${theme.media.lteSmall} {
      margin-left: 0;
    }
  `}
`;
