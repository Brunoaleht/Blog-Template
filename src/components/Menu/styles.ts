import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuVisible = {
  visible: boolean;
};
const containerChange = (visible: MenuVisible['visible'], theme: DefaultTheme) => css`
  left: ${visible ? '0rem' : '-30rem'};
  overflow-y: ${visible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${visible ? '0rem' : '-32rem'};
  }
`;
export const ContainerMenu = styled.div<MenuVisible>`
  ${({ theme, visible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${containerChange(visible, theme)}
  `}
`;
export const MenuContainer = styled.nav<MenuVisible>`
  ${() => css`
    margin: auto;
    width: 100%;
  `}
`;
export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xxlarge};
      img {
        border: 0.5rem solid ${theme.colors.secondary};
      }
    }
  `}
`;
export const List = styled.ul`
  ${() => css`
    list-style: none;
    padding: 0;
  `}
`;
export const Item = styled.li``;
export const Button = styled.button<MenuVisible>`
  ${({ theme, visible }) => css`
    z-index: 2;
    position: fixed;
    top: 2rem;
    width: 4rem;
    height: 4rem;
    transition: all 300ms ease-in-out;
    left: ${visible ? '26rem' : '1rem'};
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
    > svg:hover {
      color: ${theme.colors.secondary};
    }
    @media ${theme.media.lteMedium} {
      left: ${visible ? '26rem' : '-0.4rem'};
    }
  `}
`;
