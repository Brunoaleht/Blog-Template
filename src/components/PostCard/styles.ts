import styled, { css } from 'styled-components';

import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: ${theme.colors.white};
    box-shadow: 0 0 10px rgb(0, 0, 0, 0.2);
    border-radius: 8px;
    padding: 1rem;
    cursor: pointer;
    transition: transform 100ms ease-in-out;
    ${HeadingStyles} {
      margin: ${theme.spacings.xsmall} 0 ${theme.spacings.small} 0;
    }
    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }
    &:hover {
      transform: scale(1.05);
    }
    &:hover a,
    &:hover img {
      color: ${theme.colors.secondary};
      opacity: 0.8;
    }
  `}
`;
export const Cover = styled.img`
  ${() => css`
    width: 100%;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css`
    margin-top: 0;
  `}
`;
