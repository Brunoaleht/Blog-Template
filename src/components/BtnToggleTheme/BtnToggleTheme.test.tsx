import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { BtnToggleTheme } from '.';
import { theme } from '../../styles/theme';

describe('<BtnToggleTheme />', () => {
  beforeEach(() => {
    localStorage.setItem('theme', JSON.stringify({ ...theme, name: 'inverted' }));
  });
  afterEach(() => {
    localStorage.removeItem('theme');
  });
  it('should render input without theme', () => {
    localStorage.removeItem('theme');
    renderTheme(<BtnToggleTheme />);
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });
  it('should render input with default theme', () => {
    localStorage.setItem('theme', JSON.stringify({ ...theme, name: 'default' }));
    renderTheme(<BtnToggleTheme />);
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });
  it('should render input checked with inverted theme', () => {
    renderTheme(<BtnToggleTheme />);
    expect(screen.getByRole('checkbox')).toBeChecked();
  });
  it('should render input and toggle between theme toggling input', () => {
    const { container } = renderTheme(<BtnToggleTheme />);
    const label = screen.getByLabelText('Toggle light and dark mode');
    const input = screen.getByRole('checkbox');

    fireEvent.click(label);
    expect(input).not.toBeChecked();
    fireEvent.click(label);
    expect(input).toBeChecked();

    expect(container).toMatchSnapshot();
  });
});
