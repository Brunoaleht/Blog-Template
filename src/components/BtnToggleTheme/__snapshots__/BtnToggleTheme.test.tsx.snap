// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BtnToggleTheme /> should render input and toggle between theme toggling input 1`] = `
.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  overflow: hidden;
  font-size: 0;
  color: rgba(0,0,0,0);
}

.c5:checked + .c3 {
  background: #dc143c;
}

.c5:focus + .c3 {
  border: solid 1px #dc143c;
}

.c5:checked + .c3::before {
  -webkit-transform: translateX(1.95rem);
  -ms-transform: translateX(1.95rem);
  transform: translateX(1.95rem);
  background: #000000;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked + .c3 {
  background: #dc143c;
}

.c2:focus + .c3 {
  border: solid 1px #dc143c;
}

.c2:checked + .c3::before {
  -webkit-transform: translateX(1.95rem);
  -ms-transform: translateX(1.95rem);
  transform: translateX(1.95rem);
  background: #ffffff;
}

.c4 {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  cursor: pointer;
  background: #f9f9f9;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border: solid 1px #ccc;
  border-radius: 2rem;
}

.c4::before {
  content: '';
  position: absolute;
  height: 2.1rem;
  width: 2.1rem;
  left: 0.4rem;
  bottom: 0.1rem;
  background: #f9f9f9;
  border-radius: 50%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      Toggle light and dark mode
      <input
        class="c2"
        type="checkbox"
        value="Dark mode active"
      />
      <span
        class="c3 c4"
      />
    </label>
  </div>
</div>
`;
